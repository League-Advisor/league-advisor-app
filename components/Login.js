import Link from "next/link";
import Image from "next/image";


export default function LoginForm({ login }) {
  return (
    <div className="absolute flex w-2/3 bg-gray-300 rounded-3xl ">
      <div
        className="flex w-full rounded z-1 color-gray-200"
        style={{ width: "100%" }}
      >
        <video
          playsInline
          autoPlay
          loop
          muted
          className="absolute w-full z-1 rounded-3xl "
          layout="fill"
          objectFit="cover"
          src="/Galio.webm"
        />

        <div className="z-0 flex w-1/2 py-10">
          <form
            className="w-full"
            onSubmit={(event) => {
              event.preventDefault(),
                login(event.target.userName.value, event.target.password.value);
            }}
          >
            <div className="z-0 justify-start w-1/3 text-xl font-bold ">
              <h2 className="z-0 justify-start text-3xl font-bold text-white">
                Login
              </h2>
            </div>
            <div className="px-5 py-5">
              <div className="items-start py-5 pl-5">
                <label
                  className="z-0 block w-1/3 my-2 mt-5 text-xl font-bold tracking-wide text-white "
                  for="userName"
                >
                  username
                </label>
              </div>
              <div className="px-10">
                <input
                  className="z-0 block w-full px-4 py-4 leading-tight text-gray-700 border border-gray-200 border-solid rounded appearance-none bg-white/90 focus:outline-none focus:bg-white focus:border-gray-200 "
                  id="userName"
                  type="text"
                  placeholder="username"
                  required
                />
              </div>

              <div className="items-start py-5 pl-5">
                <label
                  className="z-0 block w-1/3 my-2 mt-5 text-xl font-bold tracking-wide text-white "
                  for="password"
                >
                  password
                </label>
              </div>
              <div className="z-0 px-10 ">
                <input
                  className="z-0 block w-full px-4 py-4 leading-tight text-black border border-gray-200 border-solid rounded appearance-none bg-white/90 focus:outline-none focus:bg-white focus:border-gray-200"
                  id="password"
                  type="password"
                  placeholder="password"
                  required
                />
              </div>

              <div className="z-0 px-10 py-20">
                <input
                  className="block w-full px-4 py-4 text-xl font-bold leading-tight text-gray-800 transition bg-white rounded hover:bg-gray-300 active:bg-gray-500"
                  type="submit"
                  value="login"
                  required
                />
              </div>
            </div>
          </form>
        </div>
        <div className="z-0 flex justify-center w-2/3 py-10 ">
          <div className="z-0 w-5/6 px-5 bg-gray-100/90 rounded-xl ">
            <div className="z-0 justify-start p-5 ">
              <h2 className="z-0 py-5 text-3xl font-bold ">
                Welcome to League Advisor
              </h2>
              <br />

              <p className="text-xl font-bold">
                League Advisor is your perfect League of Legends companion!
                <br /> <br />
                It is deigned to enhance your experience with League of Legends
                and help you get more into the game regardless of your rank or
                experience. Stay up to date with game news, explore game
                champions, items and find the prefect items build for any
                situation, and more!
              </p>
              <br />
              <br />
              <h4 className="text-xl font-bold">
                Not a member already?
                <Link href="#">
                  <a className="text-blue-500 hover:text-red"> Sign up here!</a>
                </Link>
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}


