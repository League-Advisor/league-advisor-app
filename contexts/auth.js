import { createContext, useContext, useState } from 'react';
import jwt from 'jsonwebtoken';
import axios from 'axios'
const baseUrl = process.env.NEXT_PUBLIC_API_URL;
const tokenUrl = baseUrl + '/api/token/';
import { useEffect } from "react";

const AuthContext = createContext();

export function useAuth() {
    const auth = useContext(AuthContext);
    if (!auth) throw new Error('You forgot AuthProvider!');
    return auth;
}

export function AuthProvider(props) {

    const [state, setState] = useState({
        tokens: null,
        user: null,
        login: login,
    });

    async function login(username, password) {

        const response = await axios.post(tokenUrl, { username, password });

        const decodedAccess = jwt.decode(response.data.access);

        const newState = {
            tokens: response.data,
            user: {
                id: decodedAccess.user_id,
                username: decodedAccess.username,
                email: decodedAccess.email,
                summoner_name: decodedAccess.summoner_name,
                profile_icon: decodedAccess.profile_icon,
                summoner_server: decodedAccess.summoner_server,
                summoner_level: decodedAccess.summoner_level,
                summoner_rank: decodedAccess.summoner_rank,
                summoner_champion_mastery: decodedAccess.summoner_champion_mastery,
                summoner_match_history: decodedAccess.summoner_match_history,
            },

        }

        localStorage.setItem("token", JSON.stringify(newState))
        setState(prevState => ({ ...prevState, ...newState }));
    }

    useEffect(() => {
        const data = localStorage.getItem("token")
        if (data) {
            setState(JSON.parse(data))
        }
    }, {})

    return (
        <AuthContext.Provider value={state}>
            {props.children}
        </AuthContext.Provider>
    );
}