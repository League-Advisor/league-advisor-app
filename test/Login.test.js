import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import * as React from 'react'
import { useAuth } from '../contexts/auth'
import userEvent from '@testing-library/user-event'



const Providers = ({ children }) => {
  return children;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };





import Login from '../components/Login'




describe("LoginRendering", () => {
  it("should render", () => {
    const onSubmitSpy = jest.fn();
    render(<Login onSubmit={onSubmitSpy} />);

    const video = screen.findByTitle(/video/i)

    const div = screen.findByTitle(/div/i)

    const form = screen.findByTitle(/form/i)

    const input = screen.findByTitle(/input/i)

    const image = screen.findByTitle(/Image/i)

    const p = screen.findByTitle(/p/i)

    const h4 = screen.findByTitle(/h4/i)


    expect(video).toBeInTheDocument
    expect(div).toBeInTheDocument
    expect(form).toBeInTheDocument
    expect(input).toBeInTheDocument
    expect(image).toBeInTheDocument
    expect(p).toBeInTheDocument
    expect(h4).toBeInTheDocument

  });
});

describe("Login", () => {
  it("should submit on submit", () => {
    const onSubmitSpy = jest.fn();
    render(<Login onSubmit={onSubmitSpy} />);

    onSubmitSpy({ "password": "admin", "username": "admin" })

    expect(onSubmitSpy).toHaveBeenLastCalledWith({
      password: 'admin',
      username: 'admin',
    });
  });
});


describe("LoginForm", () => {
  it("should accept filling the forms", () => {
    const onSubmitSpy = jest.fn();
    render(<Login onSubmit={onSubmitSpy} />);

    fireEvent.change(screen.getByLabelText(/username/i), {
      target: { value: 'username' },
    })
    const username = screen.getByLabelText(/username/i)

    fireEvent.change(screen.getByLabelText(/password/i), {
      target: { value: 'password' },
    })
    const password = screen.getByLabelText(/password/i)

    expect(username.value).toEqual('username');
    expect(password.value).toEqual('password');

  });
});




import renderer from 'react-test-renderer';


it('Login renders correctly', () => {
  const login = renderer
    .create(<Login />)
    .toJSON();
  expect(login).toMatchSnapshot();
});