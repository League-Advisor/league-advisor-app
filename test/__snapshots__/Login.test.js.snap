// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="absolute flex w-full min-h-screen text-center bg-gray-300"
>
  <div
    className="flex w-full rounded z-1 color-gray-200"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <video
      autoPlay={true}
      className="absolute w-full z-1"
      layout="fill"
      loop={true}
      muted={true}
      objectFit="cover"
      playsInline={true}
      src="/Galio.webm"
    />
    <div
      className="z-0 flex w-1/2 py-20"
    >
      <form
        className="w-full py-20"
        onSubmit={[Function]}
      >
        <div
          className="z-0 justify-start w-1/3 text-xl font-bold "
        >
          <h2
            className="z-0 justify-start text-5xl font-bold text-white"
          >
            Login
          </h2>
        </div>
        <div
          className="px-5 py-5"
        >
          <div
            className="items-start py-5 "
          >
            <label
              className="z-0 block w-1/3 my-2 mt-5 text-xl font-bold tracking-wide text-white "
              htmlFor="userName"
            >
              username
            </label>
          </div>
          <div
            className="px-10"
          >
            <input
              className="z-0 block w-full px-4 py-4 leading-tight text-gray-700 border border-gray-200 border-solid rounded appearance-none bg-white/80 focus:outline-none focus:bg-white focus:border-gray-200 "
              id="userName"
              placeholder="username"
              required={true}
              type="text"
            />
          </div>
          <div
            className="items-start py-5 "
          >
            <label
              className="z-0 block w-1/3 my-2 mt-5 text-xl font-bold tracking-wide text-white "
              htmlFor="password"
            >
              password
            </label>
          </div>
          <div
            className="z-0 px-10 "
          >
            <input
              className="z-0 block w-full px-4 py-4 leading-tight text-black border border-gray-200 border-solid rounded appearance-none bg-white/90 focus:outline-none focus:bg-white focus:border-gray-200"
              id="password"
              placeholder="password"
              required={true}
              type="password"
            />
          </div>
          <div
            className="z-0 px-10 py-20"
          >
            <input
              className="block w-full px-4 py-6 text-2xl font-bold leading-tight text-gray-800 transition rounded bg-white/90 hover:bg-gray-300/90 active:bg-gray-500/80 active:text-white"
              required={true}
              type="submit"
              value="login"
            />
          </div>
        </div>
      </form>
    </div>
    <div
      className="z-0 flex justify-center w-2/3 py-20 "
    >
      <div
        className="z-0 w-5/6 px-5 py-20 bg-gray-100/90 rounded-xl"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Flogo.png&w=640&q=75"
              srcSet="/_next/image?url=%2Flogo.png&w=384&q=75 1x, /_next/image?url=%2Flogo.png&w=640&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
        <div
          className="z-0 justify-start p-5 "
        >
          <h2
            className="z-0 py-5 text-4xl font-bold "
          >
            Welcome to League Advisor
          </h2>
          <br />
          <p
            className="text-xl font-bold"
          >
            League Advisor is your perfect League of Legends companion!
            <br />
             
            <br />
            It is designed to enhance your experience with League of Legends and help you get more into the game regardless of your rank or experience. Stay up to date with game news, explore game champions, items and find the prefect items build for any situation, and more!
          </p>
          <br />
          <br />
          <h4
            className="text-xl font-bold"
          >
            Not a member already?
            <button
              onClick={[Function]}
              type="button"
            >
              <a
                className="pl-5 text-2xl font-bold text-blue-500 hover:text-red-500"
              >
                 Sign up here!
              </a>
            </button>
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
